turbopack/crates/turbopack-ecmascript/src/references/esm/export.rs

```diff
@@ -663,31 +690,62 @@ impl EsmExports {
-                                        DUMMY_SP,
-                                        Default::default()
-                                    ),
-                                )
-                                ]
-                            } else {
-                                smallvec![quote!(
-                                    "(() => $expr)" as Expr,
-                                    expr: Expr = expr.map_either(Expr::from, Expr::from).into_inner()
-                                )]
+                                                        let expr = ident.as_expr_individual(DUMMY_SP);
+                            let read_expr = expr.map_either(Expr::from, Expr::from).into_inner();
+                            use crate::references::esm::base::ReferencedAssetIdent;
+                            match &ident {
+                                ReferencedAssetIdent::LocalBinding {ctxt, liveness,.. } => {
+                                    debug_assert!(*mutable == (*liveness == Liveness::Mutable), "If the re-export is mutable, the merged local must be too");
+                                    // If we are re-exporting something but got merged with it we can treat it like a local export
+                                     match (liveness, export_usage_info.is_circuit_breaker) {
+                                        (Liveness::Constant, false) => {
+                                            ExportBinding::Value(read_expr)
+                                        }
+                                        // If the value might change or we are a circuit breaker we must bind a
+                                        // getter to avoid capturing the value at the wrong time.
+                                        (Liveness::Live, _) | (Liveness::Constant, true) => {
+                                            // In the constant case, we could still export as a value if we knew that the module
+                                            // came _before_ us, but we don't at this point.
+                                            ExportBinding::Getter(quote!("() => $local" as Expr, local: Expr = read_expr))
+                                        }
+                                        (Liveness::Mutable, _) => ExportBinding::GetterSetter(
+                                            quote!("() => $local" as Expr, local: Expr= read_expr.clone()),
+                                            quote!(
+                                                "($new) => $lhs = $new" as Expr,
+                                                lhs: AssignTarget = AssignTarget::Simple(
+                                                        ident.as_expr_individual(DUMMY_SP).map_either(|i| SimpleAssignTarget::Ident(i.into()), SimpleAssignTarget::Member).into_inner()),
+                                                new = Ident::new(format!("new_{name}").into(), DUMMY_SP, *ctxt),
```

Comment:
Nit: move long expressions outside of the macro so that rustfmt can operate over them.

rustfmt can only handle function-style macros where every argument is a valid expression.